R/save.R

```diff
@@ -0,0 +1,66 @@
+#' Save TensorFlow Model
+#' 
+#' Saves a TensorFlow model with the goal of deploying and reusing across systems.
+#' 
+#' @param model A TensorFlow model. For instance, as generated by \code{linear_regressor()}.
+#' @param path The destination path where the model will be saved.
+#' @param spec The model specification, each model has a default specification;
+#'   but it can be further defined specific functions like \code{tf_save_regressor_spec}.
+#'   
+#' @family save
+#' @export
+tf_save <- function(model, path, spec) {
+  UseMethod("tf_save")
+}
+
+#' Create Save-Specification from Model
+#' 
+#' Creates default specification for a given model.
+#' 
+#' @param model A TensorFlow model. For instance, as generated by \code{linear_regressor()}.
+#' 
+#' @export
+#' @family save
+tf_save_spec_from_model <- function(model) {
+  UseMethod("tf_save_spec_from_model")
+}
+
+#' @export
+#' @family save
+tf_save_spec_from_model.tf_estimator <- function(model) {
+  feature_columns <- c(
+    model$params$feature_columns,
+    model$params$linear_feature_columns,
+    model$params$dnn_feature_columns
+  )
+  
+  list(
+    feature_columns = model$params$feature_columns,
+    weight_column = model$params$weight_column,
+    label_key = "label"
+  )
+}
+
+#' @export
+#' @family save
+tf_save.tf_estimator_regressor <- function(model, path, spec = tf_save_spec_from_model(model)) {
```

Comment:
Regressor -> regression so it's more consistent with tf_estimator_classification?

Reply:
Another issue is that we have to error out like "this is not supported for custom estimators" since for custom estimators you'd have to construct your feature spec like this (as I mentioned in the channel):
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/estimator/estimator_test.py#L1475-L1478